### DISCUSSION TITLE
What causes the great difference between span time and profiling time

### DISCUSSION THREAD

**Discussion by xlq20080808** [apache/skywalking#9574] [Q&A]
JDK:1.8
Skywalking oap: 9.1.0
Skywalking-Javaagent:8.11.0

I want to troubleshoot problems through the profiling function,the execution time of span differs greatly from that of profiling
<img width="1210" alt="image" src="https://user-images.githubusercontent.com/33999699/188776537-3aa19f3c-dcdc-4e5b-ae2c-c99996690f3b.png">

I've tried it many times, but it's all this.My understanding is that there should not be a big difference between these two times,what causes this phenomenon,in other words, what link may cause the difference between the two times

My test method directly returns the result, and there is no other logic

**Comment by wu-sheng** [comment]
https://skywalking.apache.org/docs/main/latest/en/concepts-and-designs/sdk-profiling/

Learn how SDK profiling work. This would not hard to understand.
If the profiling thread is not scheduled, due to JVM doesn't guarantee the thread priority, it would miss the time of the execution and show you shorter duration, if it is not a bug.
Generally, besides bugs, it is fine to present a shorter time, the profiling is a method helping you to locate bottlenecks of method, rather than proving span duration is accurate or not. Those are two methods to inspect things from two perspectives.

**Reply by wu-sheng** [reply]
You should check the reported snapshots, to find out whether it is miscalculated, or only few snapshots reported.
